store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/operation/addelements/handler/AddElementsFromRDDHandler.java

```diff
@@ -0,0 +1,138 @@
+                final SparkSession spark = ((SparkUser) user).getSparkSession();
+                configureSpark(spark, store.getProperties());
+                // Write the data out
+                LOGGER.info("Starting to write the unsorted Parquet data to " + store.getProperties().getTempFilesDir() + " split by group");
+                final WriteUnsortedDataFunction writeUnsortedDataFunction = new WriteUnsortedDataFunction(store.getSchemaUtils(), store.getProperties());
+                operation.getInput().foreachPartition(writeUnsortedDataFunction);
+                LOGGER.info("Finished writing the unsorted Parquet data to " + store.getProperties().getTempFilesDir());
+                // Spark read in the data, aggregate and sort the data
+                LOGGER.info("Starting to write the sorted and aggregated Parquet data to " + store.getProperties().getDataDir() + " split by group");
+                new AggregateAndSortTempData(store, spark);
+                LOGGER.info("Finished writing the sorted and aggregated Parquet data to " + store.getProperties().getDataDir());
+                // Generate the file based index
+                LOGGER.info("Starting to write the indexes");
+                new GenerateIndices(store);
+                LOGGER.info("Finished writing the indexes");
+            } else {
+                throw new OperationException("This operation requires the user to be of type SparkUser.");
+            }
+        } catch (IOException e) {
+            throw new OperationException("IO Exception: Failed to connect to the file system", e);
+        } catch (StoreException e) {
+            throw new OperationException(e.getMessage(), e);
+        }
+        try {
+            final FileSystem fs = store.getFS();
+            Path tempDir = new Path(store.getProperties().getTempFilesDir());
+            // move data from temp to data
+            final long snapshot = System.currentTimeMillis();
+            final String destPath = store.getProperties().getDataDir() + "/" + snapshot + "/";
+            fs.mkdirs(new Path(destPath));
+            fs.rename(new Path(store.getProperties().getTempFilesDir() + "/sorted/graph"), new Path(destPath + "graph"));
+            final Path tempReversePath = new Path(store.getProperties().getTempFilesDir() + "/sorted/reverseEdges");
+            if (fs.exists(tempReversePath)) {
+                fs.rename(tempReversePath, new Path(destPath + "reverseEdges"));
+            }
+            // set the data dir property
+            store.setCurrentSnapshot(snapshot);
+            // reload indices
+            store.loadIndices();
+            fs.delete(tempDir, true);
+            LOGGER.info("Temp data directory '" + tempDir.toString() + "' has been deleted.");
+            while (fs.listStatus(tempDir.getParent()).length == 0) {
+                tempDir = tempDir.getParent();
+                LOGGER.info("Empty directory '" + tempDir.toString() + "' has been deleted.");
```

Comment:
Bit picky but perhaps put this log statement after the directory has actually been deleted.